modules/raft/src/main/java/org/apache/ignite/raft/jraft/core/NodeImpl.java

```diff
@@ -606,7 +610,50 @@ private void handleElectionTimeout() {
             }
         }
     }
-
+    
+    /**
+     * Method that adjusts election timeout after several consecutive unsuccessful leader elections.
+     *
+     * Notes about general algorithm: the main idea is that in a stable cluster election timeout should be relatively small, but when cluster
```

Comment:
Raft doesn't know a bit about some "membership layer". The javadoc should be rewritten in terms of a raft node stability, like long GC pauses or OS freezes, preventing elections from a completion. Also, I would rewrite this to using strategy pattern, similar Ignite 2 org.apache.ignite.spi.TimeoutStrategy. The current election tmieout should be the function of unsuccessful election rounds. Also it would be good to print election rounds in the log. This strategy must be configurable by NodeOptions. By default, noop strategy must be used.